### PR TITLE
Docs: added documentation for the "prepare time series"-transformation.

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
This will add documentation on how the "prepare time series"-transformation is working and can be used to resolve the issue with panels not supporting all the time series data frame formats.

### CODE REVIEW

File: docs/sources/developers/plugins/data-frames.md [hunk 1]
```diff
@@ -163,6 +163,8 @@ Dimensions: 5 fields by 2 rows
 +---------------------+------------------+------------------+------------------+------------------+
 ```
 
+> **Note:** All panels does not support the wide time series data frame format. To keep full backward compatibility we have introduced a transformation that can be used to convert from the wide to the long format. Read more about how to use it here: [Prepare time series-transformation]({{< relref "../../panels/transformations/types-options.md#prepare-time-series" >}}).
```
Comment (jackw) [inline docs/sources/developers/plugins/data-frames.md:166]:
```suggestion
> **Note:** Not all panels support the wide time series data frame format. To keep full backward compatibility we have introduced a transformation that can be used to convert from the wide to the long format. Read more about how to use it here: [Prepare time series-transformation]({{< relref "../../panels/transformations/types-options.md#prepare-time-series" >}}).
```

File: docs/sources/panels/transformations/types-options.md [hunk 2]
```diff
@@ -437,3 +435,15 @@ In the following example, we are stripping the prefix from event types. In the b
 With the transformation applied, you can see we are left with just the remainder of the string.
 
 {{< figure src="/static/img/docs/transformations/rename-by-regex-after-7-3.png" class="docs-image--no-shadow" max-width= "1100px" >}}
+
+## Prepare time series
+
+> **Note:** This transformation is available in Grafana 7.5.10+ and Grafana 8.0.6+.
+
+You might end up in a situation where a data source returns time series data in a time series format that the panel you want to use does not support ([read more about the different data frame formats here]({{< relref "../../developers/plugins/data-frames.md" >}})).
```
Comment (jackw) [inline docs/sources/panels/transformations/types-options.md:443]:
```suggestion
You might end up in a situation where a data source returns time series data in a format that isn't supported by the panel you want to use. [Read more about the different data frame formats here]({{< relref "../../developers/plugins/data-frames.md" >}}).
```

File: docs/sources/panels/transformations/types-options.md [hunk 3]
```diff
@@ -23,60 +23,60 @@ Grafana comes with the following transformations:
...
 Query B:
 
-| AQI     | Errors |
-| ------- | ------ |
-| 3.2     | 5      |
-
-
+| AQI | Errors |
+| --- | ------ |
+| 3.2 | 5      |
```
Comment (dprokop) [inline docs/sources/panels/transformations/types-options.md:79]: not related? :)

**Review by jackw — APPROVED** [review]
LGTM! 🚀 

I'm no content writer but left a couple of suggestions. Take them with a pinch of salt.🧂 

It looks like prettier ran on the files. Hopefully those changes don't affect how it looks on the website.

File: docs/sources/panels/transformations/types-options.md (hunk 2 above)
Comment (dprokop) [inline docs/sources/panels/transformations/types-options.md:443]:
```suggestion
Prepare time series transformation is useful when a data source returns time series data in a format that the panel you want to use does not support ([read more about the different data frame formats here]({{< relref "../../developers/plugins/data-frames.md" >}})).
```

**Review by dprokop — APPROVED** [review]
Looks good! Plenty of unrelated changes, but the essence is good 🚀

File: docs/sources/panels/transformations/types-options.md [hunk 4]
```diff
@@ -23,60 +23,60 @@ Grafana comes with the following transformations:
...
 - [Series to rows]({{< relref "./types-options.md#series-to-rows" >}})
 - [Sort by]({{< relref "./types-options.md#sort-by" >}})
+- [Prepare-time-series]({{< relref "./types-options.md#prepare-time-series" >}})
 
 Keep reading for detailed descriptions of each type of transformation and the options available for each, as well as suggestions on how to use them.
 
 ## Reduce
 
-The _Reduce_ transformation will apply a calculation to each field in the frame and return a single value.  Time fields are removed when applying
+The _Reduce_ transformation will apply a calculation to each field in the frame and return a single value. Time fields are removed when applying
```
Comment (marcusolsson) [inline docs/sources/panels/transformations/types-options.md:32]: Change from passive to active voice:

> This transformation removes any time fields when applied.

Comment (marcusolsson) [inline docs/sources/panels/transformations/types-options.md:32]:
```suggestion
The _Reduce_ transformation applies a calculation to each field in the frame and return a single value. Time fields are removed when applying
```

File: docs/sources/panels/transformations/types-options.md [hunk 5]
```diff
@@ -437,3 +435,15 @@ In the following example, we are stripping the prefix from event types. In the b
...
 
 {{< figure src="/static/img/docs/transformations/rename-by-regex-after-7-3.png" class="docs-image--no-shadow" max-width= "1100px" >}}
+
+## Prepare time series
+
+> **Note:** This transformation is available in Grafana 7.5.10+ and Grafana 8.0.6+.
+
+You might end up in a situation where a data source returns time series data in a time series format that the panel you want to use does not support ([read more about the different data frame formats here]({{< relref "../../developers/plugins/data-frames.md" >}})).
+
+This transformation will help you resolve this issue by converting the time series data from either the wide format to the long format or the other way around.
```
Comment (marcusolsson) [inline docs/sources/panels/transformations/types-options.md:445]:
```suggestion
This transformation helps you resolve this issue by converting the time series data from either the wide format to the long format or the other way around.
```

File: docs/sources/panels/transformations/types-options.md (hunk 3 above)
Comment (mckn) [reply docs/sources/panels/transformations/types-options.md:79]: Yeah, it is prettier that formats the document. Maybe I should revert those?

**Pull request merged**